src/basics/clue-result.js

```diff
@@ -39,17 +39,48 @@ export function elim_result(player, hypo_player, hand, list) {
 
 /**
  * @param  {Game} game
+ * @param  {Game} hypo_game
  * @param  {Player} hypo_player
+ * @param  {number} giver
  * @param  {number} target
  * @param  {number} focus_order
  */
-export function bad_touch_result(game, hypo_player, target, focus_order = -1) {
-	const { me, state } = game;
-	let bad_touch = 0, trash = 0;
+export function bad_touch_result(game, hypo_game, hypo_player, giver, target, focus_order = -1) {
+	const { me, state } = hypo_game;
+	let bad_touch = 0, trash = 0, avoidable_dupe = 0;
+
+	const dupe_scores = game.players.map((player, pi) => {
+		if (pi == target)
+			return Infinity;
+		let possible_dupe = 0;
+		// Check if the giver may have a touched duplicate card.
+		// TODO: Should we consider chop moved cards?
+		for (const card of state.hands[target]) {
+			const identity = card.identity();
+			// TODO: Should we cluing cards where receiver knows they are duplicated?
+			if (!identity || !hypo_game.state.isPlayable(identity))
+				continue;
+			for (const giverCard of state.hands[pi]) {
+				// Allow known duplication since we can discard to resolve it.
+				if (hypo_player.thoughts[giverCard.order].clued &&
+					game.players[pi].thoughts[giverCard.order].inferred.length > 1 &&
+					game.players[pi].thoughts[giverCard.order].inferred.has(identity))
+					possible_dupe++;
+			}
+		}
+		return possible_dupe;
+	});
+	const min_dupe = Math.min(...dupe_scores);
 
 	for (const card of state.hands[target]) {
-		// Ignore cards that aren't newly clued, focused card can't be bad touched
-		if (!card.newly_clued || card.order === focus_order)
+		// Ignore cards that aren't newly clued
+		if (!card.newly_clued)
+			continue;
+
+		avoidable_dupe = dupe_scores[giver] - min_dupe;
```

Comment:
Shouldn't this also be affected by whether the giver is touching a card that could be duplicated? Even if they may have a lot of potentially duplicated cards, as long as they aren't touching those, I don't think the clue value should be penalized.

Also, I'm not sure why this is inside the loop, since it doesn't seem to depend on any loop variables.

Reply:
Moved outside the loop, and yes I added the newly_clued check to the calculation of avoidable dupe.